frame/staking/src/lib.rs

```diff
@@ -3096,82 +3082,9 @@ impl<T, Reporter, Offender, R, O> ReportOffence<Reporter, Offender, O>
 	}
 }
 
-/// Disallows any transactions that change the election result to be submitted after the election
-/// window is open.
-#[derive(Encode, Decode, Clone, Eq, PartialEq)]
-pub struct LockStakingStatus<T>(sp_std::marker::PhantomData<T>);
-
-impl<T: Trait + Send + Sync> sp_std::fmt::Debug for LockStakingStatus<T> {
-	fn fmt(&self, f: &mut sp_std::fmt::Formatter) -> sp_std::fmt::Result {
-		write!(f, "LockStakingStatus")
-	}
-}
-
-impl<T> LockStakingStatus<T> {
-	/// Create new `LockStakingStatus`.
-	pub fn new() -> Self {
-		Self(sp_std::marker::PhantomData)
-	}
-}
-
-impl<T> Default for LockStakingStatus<T> {
-	fn default() -> Self {
-		Self::new()
-	}
-}
-
-impl<T: Trait + Send + Sync> SignedExtension for LockStakingStatus<T> {
-	const IDENTIFIER: &'static str = "LockStakingStatus";
-	type AccountId = T::AccountId;
-	type Call = <T as Trait>::Call;
-	type AdditionalSigned = ();
-	type DispatchInfo = frame_support::weights::DispatchInfo;
-	type Pre = ();
-
-	fn additional_signed(&self) -> Result<(), TransactionValidityError> { Ok(()) }
-
-	fn validate(
-		&self,
-		_who: &Self::AccountId,
-		call: &Self::Call,
-		_info: Self::DispatchInfo,
-		_len: usize,
-	) -> TransactionValidity {
-		if let Some(inner_call) = call.is_sub_type() {
-			if let ElectionStatus::Open(_) = <Module<T>>::era_election_status() {
-				match inner_call {
-					Call::<T>::set_payee(..) |
-					Call::<T>::set_controller(..) |
-					Call::<T>::set_validator_count(..) |
-					Call::<T>::force_no_eras(..) |
-					Call::<T>::force_new_era(..) |
-					Call::<T>::set_invulnerables(..) |
-					Call::<T>::force_unstake(..) |
-					Call::<T>::force_new_era_always(..) |
-					Call::<T>::cancel_deferred_slash(..) |
-					Call::<T>::set_history_depth(..) |
-					Call::<T>::reap_stash(..) |
-					Call::<T>::submit_election_solution(..) |
-					Call::<T>::submit_election_solution_unsigned(..) => {
-						// These calls are allowed. Nothing.
-					}
```

Comment:
as a note, the following extrinsics are now allowed:
* bond
* bond_extra
* unbond
* withdraw_unbonded
* validate
* rebond

Reply:
* bond_extra and rebond can change slightly the score of previous solution, not sure if they should be allowed or not
* unbond shouldn't be allowed because a valid submitted solution could become invalid as far as I see.
* validate shouldn't be allowed because it removes Nominations if the stash was a nominator before

*

Reply:
right you brought this up in the previous PR as well, and I thought it was safe due to feedback from other reviews. 

You are right anything that touches `Ledger` is also banned since it changes the score.

Reply:
How about `bond` though? 

Even if you bond and validate immediately, it is fine. It is pointless since you cannot get elected for the next set because you are not in `snapshot_validators` and you cannot be indexed. 

You are right about the rest as far as I can see.

Reply:
and in this sense, all the reward stuff must also be banned, if your reward destination is `Staked`.

Reply:
> How about bond though?
>
> Even if you bond and validate immediately, it is fine. It is pointless since you cannot get elected for the next set because you are not in snapshot_validators and you cannot be indexed.

Yes it doesn't make previous solution invalid but if the offchain worker is run with this new validator, it can produce invalid solution. (I am not sure the offchain worker is run after this can happen but I think so). And producing invalid solution may not be very wanted.

But the function validate shouldn't be allowed IMO so bond is fine anyway.

Reply:
yeah bond alone should be fine.

Reply:
oh indeed I forgot payout functions and those can change the stake so modifying slightly submitted solution score.